extensions/control-plane/provision/provision-bigquery/src/test/java/org/eclipse/edc/connector/provision/gcp/BigQueryProvisionerTest.java

```diff
@@ -0,0 +1,188 @@
+                .build();
+
+        assertThat(bigQueryProvisioner.canProvision(new ResourceDefinition() {
+            @Override
+            public <RD extends ResourceDefinition, B extends Builder<RD, B>> B toBuilder() {
+                return null;
+            }
+        })).isFalse();
+    }
+
+    @Test
+    void testCanDeprovisionTrue() {
+        var bigQueryProvisioner = BigQueryProvisioner.Builder.newInstance(gcpConfiguration, monitor)
+                .build();
+
+        var provisionedResource = BigQueryProvisionedResource.Builder.newInstance()
+                .id(RESOURCE_ID)
+                .transferProcessId(TRANSFER_ID)
+                .resourceDefinitionId(RESOURCE_DEFINITION_ID)
+                .resourceName(RESOURCE_NAME)
+                .build();
+
+        assertThat(bigQueryProvisioner.canDeprovision(provisionedResource)).isTrue();
+    }
+
+    @Test
+    void testCanDeprovisionFalse() {
+        var bigQueryProvisioner = BigQueryProvisioner.Builder.newInstance(gcpConfiguration, monitor)
+                .build();
+
+        assertThat(bigQueryProvisioner.canDeprovision(new ProvisionedResource() {
+        })).isFalse();
+    }
+
+    @Test
+    void provisionSuccess() throws InterruptedException, ExecutionException {
+        var bigQueryProvisioner = BigQueryProvisioner.Builder.newInstance(gcpConfiguration, monitor)
+                .bqFactory(new BigQueryFactoryTest())
+                .build();
+
+        var resourceDefinition = BigQueryResourceDefinition.Builder.newInstance()
+                .id(RESOURCE_ID)
+                .transferProcessId(TRANSFER_ID)
+                .property(BigQueryService.PROJECT, TEST_PROJECT)
+                .property(BigQueryService.DATASET, TEST_DATASET)
+                .property(BigQueryService.TABLE, TEST_TABLE)
+                .property(BigQueryService.CUSTOMER_NAME, CUSTOMER_NAME)
+                .build();
+
+        var policy = Policy.Builder.newInstance().build();
+
+        var table = mock(Table.class);
+        when(table.exists()).thenReturn(true);
+        when(bigQuery.getTable(TEST_TARGET.getTableId())).thenReturn(table);
+
+        var response = bigQueryProvisioner.provision(resourceDefinition, policy).join();
+
+        assertThat(response.succeeded()).isTrue();
+        assertThat(response.getContent().getResource()).isInstanceOfSatisfying(BigQueryProvisionedResource.class, resource -> {
+            assertThat(resource.getId()).isEqualTo(RESOURCE_ID);
+            assertThat(resource.getTransferProcessId()).isEqualTo(TRANSFER_ID);
+            assertThat(resource.getProject()).isEqualTo(TEST_PROJECT);
+            assertThat(resource.getDataset()).isEqualTo(TEST_DATASET);
+            assertThat(resource.getTable()).isEqualTo(TEST_TABLE);
+            assertThat(resource.hasToken()).isTrue();
+            assertThat(resource.getCustomerName()).isEqualTo(CUSTOMER_NAME);
+        });
```

Comment:
all the token creation logic is not tested

Reply:
Added tests for testing the creation of the token